docs/zkapps/tutorials/anonymous-message-board.mdx

```diff
@@ -2,187 +2,198 @@
 
-```sh
-$ zk file message
-```
+  ```sh 
+  cd message-board
+  ```
+
+  For this tutorial, you run commands from the root of the `message-board` directory as you work in the `src` directory on files that contain the TypeScript code for the smart contract. 
+  
+  Each time you make updates, then build or deploy, the TypeScript code is compiled into JavaScript in the `build` directory. See the included [README](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/anonymous-message-board/README.md) file with usage instructions.
+
+### Prepare the project
+
+Example smart contract files come with the new project, you can delete them if you want.  
+
+1. Optional: Delete the example files not used in this tutorial:
+
+  ```sh
+  $ rm src/Add.ts
+  $ rm src/Add.test.ts
+  ```
+
+1. To create the `src/message.ts` file and generate the corresponding `src/message.test.ts` test file:
+
+  ```sh
+  $ zk file message
+  ```
+
+1. Copy the entire contents of the [message.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/anonymous-message-board/src/message.ts) example file into your local `message-board/src/message.ts` file. 
+
+  ```ts
+  import {
+    Field,
+    SmartContract,
+    state,
+    State,
+    method,
+    PrivateKey,
+    PublicKey,
+    Poseidon,
+  } from 'o1js';
+
+  // These private keys are exported so that experimenting with the contract is
+  // easy. Three of them (the Bobs) are used when the contract is deployed to
+  // generate the public keys that are allowed to post new messages. Jack's key
+  // is never added to the contract. So he won't be able to add new messages. In
+  // real life, we would only use the Bobs' public keys to configure the contract,
+  // and only they would know their private keys.
+
+  export const users = {
+    Bob: PrivateKey.fromBase58(
+      'EKFAdBGSSXrBbaCVqy4YjwWHoGEnsqYRQTqz227Eb5bzMx2bWu3F'
+    ),
+    SuperBob: PrivateKey.fromBase58(
+      'EKEitxmNYYMCyumtKr8xi1yPpY3Bq6RZTEQsozu2gGf44cNxowmg'
+    ),
+    MegaBob: PrivateKey.fromBase58(
+      'EKE9qUDcfqf6Gx9z6CNuuDYPe4XQQPzFBCfduck2X4PeFQJkhXtt'
+    ), // This one says duck in it :)
+    Jack: PrivateKey.fromBase58(
+      'EKFS9v8wxyrrEGfec4HXycCC2nH7xf79PtQorLXXsut9WUrav4Nw'
+    ),
+  };
+
+  export class Message extends SmartContract {
+    // On-chain state definitions
+    @state(Field) message = State<Field>();
+    @state(Field) messageHistoryHash = State<Field>();
+    @state(PublicKey) user1 = State<PublicKey>();
+    @state(PublicKey) user2 = State<PublicKey>();
+    @state(PublicKey) user3 = State<PublicKey>();
+
+    init() {
+      // Define initial values of on-chain state
+      this.user1.set(users['Bob'].toPublicKey());
+      this.user2.set(users['SuperBob'].toPublicKey());
+      this.user3.set(users['MegaBob'].toPublicKey());
+      this.message.set(Field(0));
+      this.messageHistoryHash.set(Field(0));
+    }
+
+    @method publishMessage(message: Field, signerPrivateKey: PrivateKey) {
```

Comment:
```suggestion
    @method async publishMessage(message: Field, signerPrivateKey: PrivateKey) {
```

Reply:
This will be new async circuit syntax after the next `o1js` release https://github.com/o1-labs/o1js/pull/1477.